src/mem/metaslab.h

```diff
@@ -89,20 +74,51 @@ namespace snmalloc
      */
     uintptr_t remote_and_sizeclass{0};
 
+    /**
+     * Constructor from two pointer-sized words.  The subclass is responsible
+     * for ensuring that accesses to these are type-safe.
+     */
+    constexpr MetaEntryBase(uintptr_t m, uintptr_t ras)
+    : meta(m), remote_and_sizeclass(ras)
+    {}
+
+    /**
+     * Default constructor, zero initialises.
+     */
+    constexpr MetaEntryBase() : MetaEntryBase(0, 0) {}
+
   public:
-    constexpr MetaEntry() = default;
+    /**
+     * When a meta entry is in use by the back end, it exposes tow words of
+     * state.  The low bits in both are reserved.  Bits in this bitmask must
+     * not be set by the back end in either word.
+     *
+     * During a major release, this constraint may be weakened, allowing the
+     * back end to set more bits.
+     */
+    static constexpr address_t BackendReservedMask =
+      (REMOTE_BACKEND_MARKER + 1);
```

Comment:
It might also be worth a comment that this mask applies to both words, and so (at present) is a strict superset of actually reserved bits in each word, with the need for two reserved bits arising from the desire to make sizeclass extraction shift-less (otherwise we could put the `REMOTE_BACKEND_MARKER` at the same position as `META_BOUNDARY_BIT`)?

Reply:
Comment added.  The +1 is from my forgetting how maths works.  Fixed.